packages/@lwc/integration-karma/test/light-dom/slotting/index.spec.js

```diff
@@ -119,6 +121,39 @@ describe('Slotting', () => {
         );
     });
 
+    it('should forward slots between shadow and light DOM', () => {
+        const nodes = createTestElement(
+            'x-forwarded-slot-consumer-shadow',
+            forwardedSlotConsumerShadow
+        );
+        const elm = nodes['x-forwarded-slot-consumer-shadow'];
+        expect(elm.innerHTML).toEqual(
+            process.env.NATIVE_SHADOW // native shadow doesn't output slots in innerHTML
+                ? // innerHTML in native shadow only renders the order of the slotted content but not the slot order
+                  `<x-forward-light-slot-to-shadow><x-shadow-container><p slot="lower">Upper slot content forwarded</p><p>Default slot forwarded</p><p slot="upper">Lower slot content forwarded</p></x-shadow-container></x-forward-light-slot-to-shadow>`
+                : `<x-forward-light-slot-to-shadow><x-shadow-container><slot name="upper"><p slot="upper">Lower slot content forwarded</p></slot><slot><p>Default slot forwarded</p></slot><slot name="lower"><p slot="lower">Upper slot content forwarded</p></slot></x-shadow-container></x-forward-light-slot-to-shadow>`
+        );
+
+        if (process.env.NATIVE_SHADOW) {
+            const nestedShadowSlot = elm.querySelector('x-shadow-container');
+            expect(nestedShadowSlot.shadowRoot.children.length).toBe(3);
+
+            const nestedShadowSlotChildren = extractDataIds(nestedShadowSlot);
+            const upperSlot = nestedShadowSlotChildren['upper-slot'].assignedNodes();
+            const defaultSlot = nestedShadowSlotChildren['default-slot'].assignedNodes();
+            const lowerSlot = nestedShadowSlotChildren['lower-slot'].assignedNodes();
+
+            expect(upperSlot.length).toEqual(1);
+            expect(defaultSlot.length).toEqual(1);
+            expect(lowerSlot.length).toEqual(1);
+
+            // order matches synthetic shadow
+            expect(upperSlot[0].innerHTML).toEqual('Lower slot content forwarded');
+            expect(defaultSlot[0].innerHTML).toEqual('Default slot forwarded');
+            expect(lowerSlot[0].innerHTML).toEqual('Upper slot content forwarded');
```

Comment:
These tests just ensure that when a light DOM slot is passed to a shadow DOM element the slot mapping is preserved.

It swaps the first and last elements originally being slotted.